ext/imap/php_imap.c

```diff
@@ -1559,30 +1557,28 @@ PHP_FUNCTION(imap_listscan)
 /* {{{ Get mailbox properties */
 PHP_FUNCTION(imap_check)
 {
-	zval *streamind;
-	pils *imap_le_struct;
+	zval *imap_conn_obj;
+	php_imap_object *imap_conn_struct;
 	char date[100];
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "r", &streamind) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "O", &imap_conn_obj, php_imap_ce) == FAILURE) {
 		RETURN_THROWS();
 	}
 
-	if ((imap_le_struct = (pils *)zend_fetch_resource(Z_RES_P(streamind), "imap", le_imap)) == NULL) {
-		RETURN_THROWS();
-	}
+	GET_IMAP_STREAM(imap_conn_struct, imap_conn_obj);
 
-	if (mail_ping (imap_le_struct->imap_stream) == NIL) {
+	if (mail_ping (imap_conn_struct->imap_stream) == NIL) {
 		RETURN_FALSE;
 	}
 
-	if (imap_le_struct->imap_stream && imap_le_struct->imap_stream->mailbox) {
+	if (imap_conn_struct->imap_stream && imap_conn_struct->imap_stream->mailbox) {
 		rfc822_date(date);
 		object_init(return_value);
```

Comment:
A subject for a future RFC and not something we should add here, but we should consider making this return value a formal type with declared/typed properties.

Reply:
Just to be sure, you mean adding an object e.g. ``IMAPMailboxProperties`` which would have the values as fixed properties instead of returning stdClass, correct?

Reply:
Yes. But again, something for a future project and shouldn't be mixed in with this one.

Reply:
Agreed, wouldn't want to take this longer to land.